File: python/codeeval/132_major_element.py
import sys

def me(test):
    a = test.split(',')
    l = len(a) // 2
    c = {}
    for i in a:
        if i in c:
            c[i] += 1
            if c[i] >= l:
                return i
        else:
            c[i] = 1
    return None

with open(sys.argv[1], 'r') as test_cases:
    for test in test_cases:
        print(me(test))

File: ruby/codeeval/132_major_element.rb
def me(line)
    a = line.split(',')
    l = a.length / 2
    c = {}
    a.each do |i|
        if c[i]
            c[i] += 1
            return i if c[i] >= l
        else
            c[i] = 1
        end
    end
    return 'None'
end
File.open(ARGV[0]).each_line do |line|
    puts(me(line))
end
